### PR TITLE
MOD: Changing sign convention for Upwind BC matrices.

### LINKED ISSUE #630: Inconsistent sign convention for discretization/BC matrices
Discretizations of diffusive fluxes (flow/Darcy type) and advective fluxes (transport type) use different sign conventions regarding contributions of boundary conditions. While boundary condition terms are added for the former, they have to be subtracted for the latter in order to discretize/reconstruct a flux.

For instance, using a flow discretization `flow = pp.ad.TpfaAd`, a Darcy flux would be discretized by `flow.flux * p + flow.bound_flux * bc_val`, whereas given a transport discretization `transport = pp.ad.UpwindAd`, the advective flux would be discretized by `flux * (transport.upwind * c) - transport.bound_transport_dir * (flux * bc_val) - transport.bound_transport_neu * bc_val`.

### PR DESCRIPTION
**Pull request by vlipovac**
## Proposed changes

Implementing sign changes for Upwind BC matrices to close #630 

## Types of changes

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [X] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

- [ ] The documentation is up-to-date.
- [ ] Static typing is included in the update.
- [ ] This PR does not duplicate existing functionality.
- [ ] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: src/porepy/numerics/fv/upwind.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def assemble_matrix_rhs(
                 with multiple components is only supported in Ad mode.
                 """
             )
-        rhs = div * (bc_discr_neu + bc_discr_dir * flux_mat) * bc_values
+        rhs = -div * (bc_discr_neu + bc_discr_dir * flux_mat) * bc_values
```
Comment (vlipovac) [inline src/porepy/numerics/fv/upwind.py:112]: This I am not sure if required, and how it is related to the modifications in the tests for upwinding.
Please check.

File: tests/numerics/fv/test_upwind.py [hunk 2]
```diff
@@ -216,8 +216,8 @@ def references_market():
 @pytest.fixture()
 def rhs_market():
     collection = {
-        "test_1d_darcy_flux_positive_rhs_dir": np.array([6, 0, 0]),
-        "test_1d_darcy_flux_negative_rhs_dir": np.array([0, 0, 6]),
+        "test_1d_darcy_flux_positive_rhs_dir": - np.array([6, 0, 0]),
```
Comment (OmarDuran) [inline tests/numerics/fv/test_upwind.py:219]:
```suggestion
        "test_1d_darcy_flux_positive_rhs_dir": np.array([-6, 0, 0]),
```

File: tests/numerics/fv/test_upwind.py [hunk 3]
```diff
@@ -216,8 +216,8 @@ def references_market():
...
+        "test_1d_darcy_flux_negative_rhs_dir": - np.array([0, 0, 6]),
```
Comment (OmarDuran) [inline tests/numerics/fv/test_upwind.py:220]:
```suggestion
        "test_1d_darcy_flux_negative_rhs_dir": np.array([0, 0, -6]),
```

File: src/porepy/numerics/fv/upwind.py (hunk 1 above)
Comment (OmarDuran) [reply src/porepy/numerics/fv/upwind.py]: This change has been undone and the three failing tests have been updated.

**Review by OmarDuran — APPROVED** [review]

**Pull request merged**